chain/stmgr/call.go

```diff
@@ -64,14 +69,13 @@ func (sm *StateManager) CallAtStateAndVersion(ctx context.Context, msg *types.Me
 	nvGetter := func(context.Context, abi.ChainEpoch) network.Version {
 		return v
 	}
-
-	return sm.callInternal(ctx, msg, nil, nil, stateCid, nvGetter, true, false)
+	return sm.callInternal(ctx, msg, nil, nil, stateCid, nvGetter, true, false, false)
 }
 
 //   - If no tipset is specified, the first tipset without an expensive migration or one in its parent is used.
 //   - If executing a message at a given tipset or its parent would trigger an expensive migration, the call will
 //     fail with ErrExpensiveFork.
-func (sm *StateManager) callInternal(ctx context.Context, msg *types.Message, priorMsgs []types.ChainMsg, ts *types.TipSet, stateCid cid.Cid, nvGetter rand.NetworkVersionGetter, checkGas, applyTsMessages bool) (*api.InvocResult, error) {
+func (sm *StateManager) callInternal(ctx context.Context, msg *types.Message, priorMsgs []types.ChainMsg, ts *types.TipSet, stateCid cid.Cid, nvGetter rand.NetworkVersionGetter, checkGas, applyTsMessages bool, noReExec bool) (*api.InvocResult, error) {
```

Comment:
The negative `no` here is awkward, because you further go on to negate it with `!noReExec`, so we have the first cognitive hurdle. We also have two interacting booleans, an additional cognitive hurdle, and one where the second bool can negate the first, yet another hurdle.

Maybe an enum should be used here to express the 3 cases, replacing both booleans, and then we get to name the strategies better:

```go
type execMessageStrategy int

const (
	execNoMessages         execMessageStrategy = iota // apply no prior or current tipset messages
	execAllMessages                                   // apply all prior and current tipset messages
	execSameSenderMessages                            // apply all prior messages and any current tipset messages from the same sender
)
```

Reply:
This is done.